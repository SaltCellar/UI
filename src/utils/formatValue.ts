export interface FormatOptions {
  fractionDigits?: number;
}

export type ValueFormatter = (
  value: number,
  unit: string,
  options?: FormatOptions
) => string | number;

export const formatValue: ValueFormatter = (
  value: number,
  unit: string,
  options: FormatOptions = {}
) => {
  switch (unit.toLowerCase()) {
    case 'usd':
      return formatCurrency(value, unit, options);
    default:
      return value;
  }
};

export const formatCurrency: ValueFormatter = (
  value,
  _unit,
  { fractionDigits = 2 } = {}
) => {
  return value.toLocaleString('en', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: fractionDigits,
    maximumFractionDigits: fractionDigits,
  });
};
