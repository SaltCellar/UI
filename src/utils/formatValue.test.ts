import * as format from './formatValue';

jest.spyOn(format, 'formatCurrency');

describe('formatValue', () => {
  const formatOptions: format.FormatOptions = {};
  const value = 100.11;

  test('defaults to returning the value', () => {
    const formatted = format.formatValue(value, 'unknownUnit');
    expect(formatted).toBe(value);
  });

  test('usd unit calls formatCurrency', () => {
    const unit = 'usd';
    format.formatValue(value, unit, formatOptions);
    expect(format.formatCurrency).toBeCalledWith(value, unit, formatOptions);
  });
});

describe('formatCurrency', () => {
  const value = 100.11;
  const unit = 'USD';

  test('defaults fractionDigits', () => {
    const formattedValue = format.formatCurrency(value, unit);
    expect(formattedValue).toMatchSnapshot();
  });

  test('uses specified fractionDigits', () => {
    const formattedValue = format.formatCurrency(value, unit, {
      fractionDigits: 0,
    });
    expect(formattedValue).toMatchSnapshot();
  });
});
