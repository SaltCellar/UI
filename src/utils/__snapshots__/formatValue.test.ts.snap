// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatCurrency defaults fractionDigits 1`] = `"$100.11"`;

exports[`formatCurrency uses specified fractionDigits 1`] = `"$100"`;
